public/app/plugins/datasource/elasticsearch/query_builder.js

```diff
@@ -199,29 +228,48 @@ function (queryDef) {
   };
 
   ElasticQueryBuilder.prototype.getTermsQuery = function(queryDef) {
-    var query = {
-      "size": 0,
-      "query": {
-        "filtered": {
-          "query": {
-            "query_string": {
-              "analyze_wildcard": true,
-              "query": '$lucene_query',
-            }
-          },
-          "filter": {
-            "bool": {
-              "must": [{"range": this.getRangeFilter()}]
+    var query = {};
+    if (this.esVersion >= 5) {
```

Comment:
Unless I'm missing something important, this branching may be unnecessary in favor of just adopting `bool` queries all out since:

* [`filter` queries has been running internally as `bool` queries since ES 2.0](https://www.elastic.co/blog/better-query-execution-coming-elasticsearch-2-0) so the condition should probably be at least `this.esVersion >= 2`.  Otherwise, if this is feared to cause a regression for ES 2, then ES 5 support is going to be a broken too.

* The `bool` query [appears to have been supported since 1.3](https://www.elastic.co/guide/en/elasticsearch/reference/1.3/query-dsl-bool-query.html) and [even 0.90](https://www.elastic.co/guide/en/elasticsearch/reference/0.90/query-dsl-bool-query.html).  I tried `bool` queries with ES 1.4 locally and it doesn't seem to barf.

* There's no similar branching for Annotations so if we're going to assume branching is required, Annotations will be a regression for ES < 5.